bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -193,4 +200,34 @@ actual class Scanner internal constructor(
             )
         }
     }
+
+    class ScanSettings private constructor(
+        private val allowDuplicateKeys: Boolean,
+        private val solicitedServiceUUIDsKey: List<UUID>?
+    ) {
+
+        fun parse(): Map<Any?, *> {
+            val result: MutableMap<String, Any> =
+                mutableMapOf(CBCentralManagerScanOptionAllowDuplicatesKey to allowDuplicateKeys)
+            solicitedServiceUUIDsKey?.let {
+                result[CBCentralManagerScanOptionSolicitedServiceUUIDsKey] = it
+            }
+            return result.toMap()
+        }
+
+        data class Builder(
+            // https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerscanoptionallowduplicateskey
+            var allowDuplicateKeys: Boolean = true,
```

Comment:
Default value (in iOS) is `false`, but here is `true`

Reply:
Right now Android has always default true, for xBike we need iOS to default to true else the beacons impl breaks. 

Unfortunately there is no common scan options API  we can  pass to the common Scanner. In the future I will impl that API and have Android default to false or vice versa. Not sure what we want, we can check the pros and cons together if you like (the only con seems to be that iOS uses more battery in that mode as unlike Android it does not support scan frequency params and just continuous scans, although with great care if you stop your scan after you are done this should not be a big issue)

I also need this in xBike asap but I will keep this PR open until start on next week and see if I can make the common API, if not probably I will merge this first version if you agree. 

Note: another issue I noticed is that we made our own beacon timeout imp to timeout a beacon that has not emitted for some time (thus the issue I was having with iOS that only reported them once) but both platforms offer callbacks for device lost on scan. I think an implementation around these callbacks will be more robust. This is relevant for both first only or continuous device emission scans.

Reply:
I think kaluga should behave the same on both platforms, and on Android this is on by default, AFAIK that is the reason to change it.

Of course you could argue that means Android should disable it, but that would break stuff there.

Reply:
After this PR changes, library should behave on other projects as I was before and then you can tune (update) your project with custom setup, parameters and everything you've added for your needs

Reply:
also we should just merge if we decide what to do, not keep this open.

Reply:
@avdyushin I disagree, I see the different behaviour as a bug. In fact it's one that bit you when you wrote the beacon code, because it's behaviour is essentially broken on iOS due to this.

Making this change will not actually change the API for scanning, you just get more updates (as you do on Android). If you are relying on an iOS only behaviour in common code and this breaks, that's probably for the better.

Reply:
The battery consumption also applies to android yet there it is done by default, I don't see a reason why Google or Apple is more or less right. The end goal is the unified API that lets you control these settings in a unified manner. This PR is at least a step towards that.

If you want the default to be Apple then you could implement that (or we could have two distinct classes or methods for it an deprecate the main one), but now we at least have this solution to unify the two.

I agree it's a good idea to add docs (`README`) about this, where we can also show the way in Kotlin/Native code to pass the ScanSettings and mention that normally iOS does it the other way. we could also add it to release notes (@cikpis) that the default changed.

Reply:
:+1:

Reply:
@thoutbeckers  [d77b61b](https://github.com/splendo/kaluga/pull/501/commits/d77b61bce2a0c87853a7c2b40b5b4f82f5ab1ec4) what do you think about this?

Reply:
it's a good start but please don't use "we", it's documentation, not a meeting or story. Just state something like 
> to align the behaviour across platform the `CBCentralManagerScanOptionAllowDuplicatesKey` option is enabled on iOS.

also we agreed we should show how to disable this for iOS in the README.

Reply:
@thoutbeckers  [0fc6e0d](https://github.com/splendo/kaluga/pull/501/commits/0fc6e0d26e8b05453aa77f18291a35d3fa36e031) + [0047633](https://github.com/splendo/kaluga/pull/501/commits/0047633d4c8461c454c3a3d08588e58b11c4749b)

Reply:
I made some suggestion to clarify this further but it's resolved.